src/components/DraggableList/useDraggableInPortal.ts

```diff
@@ -0,0 +1,43 @@
+import type {DraggableChildrenFn} from 'react-beautiful-dnd';
+import {useEffect, useRef} from 'react';
+import {createPortal} from 'react-dom';
+
+type DraggableInPortalProps = {
+    shouldUsePortal?: boolean;
+};
+
+export default function useDraggableInPortal({shouldUsePortal}: DraggableInPortalProps): (render: DraggableChildrenFn) => DraggableChildrenFn {
+    const element = useRef<HTMLDivElement>(document.createElement('div')).current;
+
+    useEffect(() => {
+        if (!shouldUsePortal || !element) {
+            return () => {};
+        }
+        element.style.pointerEvents = 'none';
+        element.style.position = 'absolute';
+        element.style.height = '100%';
+        element.style.width = '100%';
+        element.style.top = '0';
+
+        document.body.appendChild(element);
+
+        return () => {
+            document.body.removeChild(element);
+        };
+    }, [element, shouldUsePortal]);
+
+    if (!shouldUsePortal) {
+        return (render) => render;
+    }
+
+    return (render) => (provided, snapshot, rubric) => {
```

Comment:
NAB:

```suggestion
    return (render): DraggableChildrenFn => (provided, snapshot, rubric) => {
```

Because we have a return type defined for `useDraggableInPortal`, this additional return type declaration here is unnecessary but it might make it obvious the types of `provided`, `snapshot`, and `rubric` are defined as part of `DraggableChildrenFn`